Source/VMware.vSphereDSC/DSCResources/VDSwitchVMHost.ps1

```diff
@@ -0,0 +1,251 @@
+    }
+
+    <#
+    .DESCRIPTION
+
+    Checks if VMHosts should be added/removed from the Distributed Switch depending on the value of the Ensure property.
+    If Ensure is set to 'Present', checks if all passed VMHosts are part of the Distributed Switch.
+    If Ensure is set to 'Absent', checks if all passed VMHosts are not part of the Distributed Switch.
+    #>
+    [bool] ShouldUpdateVMHostsInDistributedSwitch($vmHosts, $distributedSwitch) {
+        if ($this.Ensure -eq [Ensure]::Present) {
+            foreach ($vmHost in $vmHosts) {
+                $addedVMHost = $vmHost.ExtensionData.Config.Network.ProxySwitch | Where-Object -FilterScript { $_.DvsName -eq $distributedSwitch.Name }
+                if ($null -eq $addedVMHost) {
+                    return $true
+                }
+            }
+        }
+        else {
+            foreach ($vmHost in $vmHosts) {
+                $removedVMHost = $vmHost.ExtensionData.Config.Network.ProxySwitch | Where-Object -FilterScript { $_.DvsName -eq $distributedSwitch.Name }
+                if ($null -ne $removedVMHost) {
+                    return $true
+                }
+            }
+        }
+
+        return $false
+    }
+
+    <#
+    .DESCRIPTION
+
+    Returns the filtered VMHosts based on the value of the Ensure property. If Ensure is set to 'Present', it returns only
+    these VMHosts that are currently not part of the Distributed Switch. The other VMHosts in the array are ignored because they are already part
+    of the Distributed Switch. If Ensure is set to 'Absent', it returns only these VMHosts that are currently part of the Distributed Switch. The other VMHosts
+    in the array are ignored because they are not part of the Distributed Switch. In both cases a warning message is shown to the user when a specific VMHost is
+    ignored.
+    #>
+    [array] GetFilteredVMHosts($vmHosts, $distributedSwitch) {
+        $filteredVMHosts = @()
+
+        if ($this.Ensure -eq [Ensure]::Present) {
+            foreach ($vmHost in $vmHosts) {
+                $addedVMHost = $vmHost.ExtensionData.Config.Network.ProxySwitch | Where-Object -FilterScript { $_.DvsName -eq $distributedSwitch.Name }
+                if ($null -ne $addedVMHost) {
+                    Write-WarningLog -Message "VMHost {0} is already added to Distributed Switch {1} and it will be ignored." -Arguments @($vmHost.Name, $distributedSwitch.Name)
+                    continue
+                }
+
+                $filteredVMHosts += $vmHost
+            }
+        }
+        else {
+            foreach ($vmHost in $vmHosts) {
+                $removedVMHost = $vmHost.ExtensionData.Config.Network.ProxySwitch | Where-Object -FilterScript { $_.DvsName -eq $distributedSwitch.Name }
+                if ($null -eq $removedVMHost) {
+                    Write-WarningLog -Message "VMHost {0} is already removed from Distributed Switch {1} and it will be ignored." -Arguments @($vmHost.Name, $distributedSwitch.Name)
```

Comment:
Message is not completely correct because it's possible VMHost to have never been managed by the switch

Reply:
Fixed.